launcher/Application.cpp

```diff
@@ -1883,3 +1883,28 @@ const QString Application::javaPath()
 {
     return m_settings->get("JavaDir").toString();
 }
+
+#if defined Q_OS_WIN32
+void Application::addQSavePath(QString path)
+{
+    QMutexLocker locker(&m_qsaveResourcesMutex);
+    m_qsaveResources.insert(path);
+}
+
+void Application::removeQSavePath(QString path)
+{
+    QMutexLocker locker(&m_qsaveResourcesMutex);
+    m_qsaveResources.remove(path);
+}
+
+bool Application::checkQSavePath(QString path)
+{
+    QMutexLocker locker(&m_qsaveResourcesMutex);
+    for (auto r : m_qsaveResources) {
+        if (path.contains(r)) {
```

Comment:
wouldn't this be safer as `startsWith` ?

Reply:
Using contains seems better, why can't you match the full thing?

Reply:
the path here is the absolute path, while the `r`(part from `m_qsaveResources`) is not guaranteed to be absolute and may be relative.

Reply:
You could resolve the path in the constructor and grab the last component of the path. I see now there isn't a way to retrieve the suffix on qsavefile though

Reply:
The path can't be retrieved from QSaveFile sadly

Reply:
ah.